### PR TITLE
add all phases, display desription as subtitle, and use default phase filters in ideaspace

### PR DESCRIPTION
**Pull request by fuzzylogic2000**

### CODE REVIEW

**Comment by fuzzylogic2000** [comment]
Now also implements the default phase filters.

File: apps/ideas/views.py [hunk 1]
```diff
@@ -299,3 +300,21 @@ class IdeaListView(filter_views.FilteredListView):
     def get_queryset(self):
         queryset = super().get_queryset().annotate_comment_count()
         return queryset
+
+    def get_context_data(self, **kwargs):
```
Comment (fuzzylogic2000) [inline apps/ideas/views.py]: Do I have to put that into the context or can I get the phase in the list view somehow?

Comment (slomo) [reply apps/ideas/views.py]: You could factor out the phase stuff into a property:

```python
@property
def active_phase(self):
     return Phase.objects.active_phases().last()
```
And than use it in your function below:

```python
if self.active_phase:
  data = self.active_phase.content().get_phase_filters(self.active_phase.module.project.pk)
```

From the template that property can be accessed as:
```
{{ view.active_phase.title }}
```

File: apps/ideas/views.py [hunk 2]
```diff
@@ -299,3 +300,21 @@ class IdeaListView(filter_views.FilteredListView):
...
+        context = super().get_context_data(**kwargs)
+        context['active_phases'] = Phase.objects.active_phases()
+        return context
+
+    def filter_kwargs(self):
+        default_kwargs = super().filter_kwargs()
+        if Phase.objects.active_phases():
+            data = (Phase.objects.active_phases()[0].content().
```
Comment (slomo) [inline apps/ideas/views.py]: This is a bit weirdly formatted and not consistent with `.copy` below.

File: tests/ideas/test_idea_export_view.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    with pytest.raises(PermissionDenied):
+        view(request)
+
+
+@pytest.mark.django_db
+def test_idea_export_view_admin(rf, admin, idea_sketch_factory,
+                                proposal_factory):
+    # the default filters without an active phase set 'status': 'winner'
+    idea_sketch = idea_sketch_factory()
+    idea_sketch.is_winner = True
```
Comment (slomo) [inline tests/ideas/test_idea_export_view.py]: Use `idea_sketch = idea_sketch_factory(is_winner=True)` and save 3 lines, but only for a short while.

File: tests/ideas/test_idea_list_view.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    assert response.status_code == 200
+
+
+@pytest.mark.django_db
+def test_idea_list_view_idea_sketch_phase(rf, admin, idea_sketch_factory,
+                                          proposal_factory, module):
+    # the default filters for the ideaSketchPhase set
+    # 'status': 'idea_sketch' and the year (project) to the active one
+    with active_phase(module, phases.IdeaSketchPhase):
+        idea_sketch_1 = idea_sketch_factory()
```
Comment (slomo) [inline tests/ideas/test_idea_list_view.py]: Also here `idea_sketch_1 = idea_sketch_factory(module=module)` does, what one would expect.

File: apps/ideas/templates/advocate_europe_ideas/idea_list.html [hunk 5]
```diff
@@ -6,6 +6,11 @@
 <div class="idealist">
     <div class="idealist-title text-center">
         <h1>{% trans "All ideas" %}</h1>
+        {% for phase in active_phases %}
```
Comment (slomo) [inline apps/ideas/templates/advocate_europe_ideas/idea_list.html]: Why not only output the subtitle of the first phase.

File: apps/ideas/phases.py [hunk 6]
```diff
@@ -1,11 +1,45 @@
...
 
-class IdeaSketchPhase(phases.PhaseContent):
+class IdeaPhase(phases.PhaseContent):
+    app = apps.IdeasConfig.label
+    weight = 10
+    view = None
+
+    module_name = _('Advocate Europe')
+
+    features = {
```
Comment (slomo) [inline apps/ideas/phases.py]: This is done by `phases.PhaseContent`

**Review by slomo — COMMENTED** [review]
Looks good so far. I left some minor comments.

**Comment by slomo** [comment]
Looks great. I will merge this.

**Pull request merged**